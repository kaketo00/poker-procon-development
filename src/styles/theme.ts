import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
