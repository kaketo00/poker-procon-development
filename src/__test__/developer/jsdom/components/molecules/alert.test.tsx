import { ChakraProvider } from '@chakra-ui/react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import type { AlertProps } from '@/components/molecules/alert';
import Alert from '@/components/molecules/alert';
import theme from '@/styles/theme';

const Component = (props: AlertProps) => (
  <ChakraProvider theme={theme}>
    <Alert {...props} />
  </ChakraProvider>
);

describe('Alert', () => {
  test('スナップショット', () => {
    const { asFragment } = render(
      <Component
        title="ゲームを作成しました。"
        description="ID: 1"
        onClose={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('アラートを閉じる', async () => {
    const onClose = jest.fn();
    const { getByRole } = render(
      <Component
        title="ゲームを作成しました。"
        description="ID: 1"
        onClose={onClose}
      />
    );

    await userEvent.click(getByRole('button', { name: 'Close' }));

    await waitFor(() => {
      expect(onClose).toHaveBeenCalled();
    });
  });
});
