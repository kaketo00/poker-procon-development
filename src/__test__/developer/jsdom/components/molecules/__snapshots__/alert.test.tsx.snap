// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert スナップショット 1`] = `
<DocumentFragment>
  <div
    class="chakra-alert css-qvvow5"
    data-status="info"
    role="alert"
  >
    <span
      class="chakra-alert__icon css-14ogjxt"
      data-status="info"
    >
      <svg
        class="chakra-icon css-qk6lof"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="css-0"
    >
      <div
        class="chakra-alert__title css-tidvy5"
        data-status="info"
      >
        ゲームを作成しました。
      </div>
      <div
        class="chakra-alert__desc css-zzks76"
        data-status="info"
      >
        ID: 1
      </div>
    </div>
    <button
      aria-label="Close"
      class="css-4abrhy"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
